### PR TITLE
[FE] Layout for Add Item Page

### PR DESCRIPTION
**Pull request by misterRenz**
## Task Link
- https://app.asana.com/0/1202718774787420/1202733801001941/f

## Commands
- npm start

## Pre-conditions
- The Frontend developer must be working in the frontend folder of the repository

## Expected Output
- It will display the Add Item Form
- It will display a button that states "Submit"
- Button will not work for now because it still has no function on [BE] but I implemented a function on it if the users press submit it will display the data of the add item on the console.log

## Notes
- To test the add item page, go to App.js and return < AddItemPage />

## Screenshots
image

### CODE REVIEW

File: frontend/src/components/button/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import "./index.css";
+
+const STYLES = ["btn--default--color"];
+
+const SIZE = ["btn--small", "btn--large"];
```
Comment (karlolee50) [inline frontend/src/components/button/index.js]: no need for this one. just pass directly as a prop

File: frontend/src/components/button/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const Button = ({ children, type, onClick, buttonStyle, buttonSize }) => {
```
Comment (karlolee50) [inline frontend/src/components/button/index.js:4]: set default value of buttonStyle and buttonSize here

File: frontend/src/components/button/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  const checkButtonStyle = STYLES.includes(buttonStyle)
+    ? buttonStyle
+    : STYLES[0];
+
+  const checkButtonSize = SIZE.includes(buttonSize) ? buttonSize : SIZE[0];
```
Comment (karlolee50) [inline frontend/src/components/button/index.js]: you can remove these

File: frontend/src/components/button/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  return (
+    <button
+      className={`btn ${checkButtonStyle} ${checkButtonSize}`}
```
Comment (karlolee50) [inline frontend/src/components/button/index.js]:
```suggestion
      className={`btn ${buttonStyle} ${buttonSize}`}
```

File: frontend/src/components/input-text-field/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  name,
+  onChange,
+  text,
+  inputFieldWidth,
+  placeholder,
+  inputFieldDisplay,
+}) => {
+  return (
+    <div className="formInput">
+      <div>
```
Comment (karlolee50) [inline frontend/src/components/input-text-field/index.js]: will the layout be broken if we remove the div?

File: frontend/src/components/input-text-field/index.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      </div>
+      <input
+        required
+        name={name}
+        onChange={onChange}
+        style={{
+          display: `${inputFieldDisplay}`,
+          width: `${inputFieldWidth}`,
+          padding: "20px",
+        }}
```
Comment (karlolee50) [inline frontend/src/components/input-text-field/index.js]: refactor this to pass just the class name like 'width-lg' and update the css file

File: frontend/src/components/inputdate/index.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <div className="date-app">
+      <label>{text}</label>
+      <div>
+        <input
+          style={{
+            display: `${inputFieldDisplay}`,
+            width: `${inputFieldWidth}`,
+            padding: "20px",
+          }}
```
Comment (karlolee50) [inline frontend/src/components/inputdate/index.js]: refer to my previous similar comment

File: frontend/src/pages/layout-add-item-page/index.css [hunk 8]
```diff
@@ -0,0 +1,25 @@
+.addItemPage {
```
Comment (karlolee50) [inline frontend/src/pages/layout-add-item-page/index.css]: please rename the 'layout-add-item-page' folder to 'add-item'

**Review by karlolee50 — CHANGES_REQUESTED** [review]
please check

**Comment by misterRenz** [comment]
Recenty made a new commit using web-editor because of newly merge from francis' branch and it had conflict with my app.js

File: frontend/src/components/input-text-field/index.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  text,
+  inputFieldWidth,
+  placeholder,
+  inputFieldDisplay,
+}) => {
+  return (
+    <div className="formInput">
+      <label>{text}</label>
+      <input
+        className="width-lg"
```
Comment (karlolee50) [inline frontend/src/components/input-text-field/index.js]: you can pass the classname as a prop. don't forget to set a default value

File: frontend/src/components/input-text-field/index.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+import ReactDOM from "react-dom/client";
+import "./index.css";
+
+const FormInput = ({
+  name,
+  onChange,
+  text,
+  inputFieldWidth,
```
Comment (karlolee50) [inline frontend/src/components/input-text-field/index.js]: remove if not being used anymore

File: frontend/src/components/input-text-field/index.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  placeholder,
+  inputFieldDisplay,
```
Comment (karlolee50) [inline frontend/src/components/input-text-field/index.js]: remove if not being used anymore

File: frontend/src/components/inputdate/index.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
+import React, { useState } from "react";
+import "./index.css";
+
+const InputDate = ({
+  text,
+  inputFieldDisplay,
+  inputFieldWidth,
```
Comment (karlolee50) [inline frontend/src/components/inputdate/index.js]: remove if not used

File: frontend/src/components/inputdate/index.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  getDateData,
+}) => {
+  const [date, setDate] = useState("");
+  return (
+    <div className="date-app">
+      <label>{text}</label>
+      <div>
+        <input
+          className="width-lg"
```
Comment (karlolee50) [inline frontend/src/components/inputdate/index.js]: same as my previous comment. pass the classname as a prop and set default value

File: frontend/src/pages/add-item/index.js [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+            onChange={(e) => setRoom(e.target.value)}
+            placeholder="Room"
+            text="Room"
+          />
+
+          <DropDown
+            label="Reserve Item For"
+            usersOptions={usersOptions}
+            selected={reserve}
+            getReserveData={(reservedata) => setReserve(reservedata)}
```
Comment (karlolee50) [inline frontend/src/pages/add-item/index.js]: is it possible to use camel casing here?
```suggestion
            getReserveData={(reserveData) => setReserve(reserveData)}
```

File: frontend/src/pages/add-item/index.js [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+          />
+
+          <div className="inline-form">
+            <InputDate
+              text="From:"
+              getDateData={(fromdatedata) => setFromDateInput(fromdatedata)}
```
Comment (karlolee50) [inline frontend/src/pages/add-item/index.js]: camel casing here as well

File: frontend/src/pages/add-item/index.js [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+            />
+            <InputDate
+              getDateData={(todatedata) => setToDateInput(todatedata)}
```
Comment (karlolee50) [inline frontend/src/pages/add-item/index.js]: use camel casing

**Review by karlolee50 — CHANGES_REQUESTED** [review]
please check

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**